pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -145,6 +146,10 @@ stdenv.mkDerivation rec {
     # This is breaking the build of any C target. This patch removes the last
     # argument if it's found to be an empty string.
     ./trim-last-argument-to-gcc-if-empty.patch
+
+    # replace hard coded non hermetic paths by NIX_DEFAULT_BAZEL_SHELL_PATH
+    # we are doing this with a patch to ensure we systematically catch changes in this line
+    ./strict_action_env.patch
```

Comment:
please call the placeholder `@strictActionEnvPath@` (or similar), and do a `substituteAll` on the patch file before applying it, instead of the substitution in `postPatch`. You might want to move the comment from down there to here:

```
      # --experimental_strict_action_env (which will soon become the
      # default, see bazelbuild/bazel#2574) hardcodes the default
      # action environment to a value that on NixOS at least is bogus.
      # So we hardcode it to something useful.
```